src/routes/users/sign-up-route.js

```diff
@@ -0,0 +1,33 @@
+const fs = require("fs");
+const path = require("path");
+
+const signUpRoute = (request, response) => {
+    if (request.method === 'POST') {
+
+        request.on('data', function (data) {
+            request.on('end', function () {
```

Comment:
do not write inner listeners in on 

move request.on('end' from request.on('data'